src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -4395,6 +4395,149 @@ public static void DerivedTypeWithDifferentOverrides()
+        Assert.StrictEqual(value.MyStringProperty, actual.MyStringProperty);     
+    }
+
+    [Fact]
+    public static void Xml_VerifyCompilationIssueOnly()
+    {
+        MemoryStream ms = new MemoryStream();
+        try
+        {
+            var serializer = new XmlSerializer(typeof(TypeWithEnumerableMembers));
+            var value = new TypeWithEnumerableMembers();
+            serializer.Serialize(ms, value);
+        }
+        catch (Exception ex)
+        {
+            Assert.Equal(ex.GetType(), typeof(InvalidOperationException));
+        }
+
+        try
+        {
+            var serializer = new XmlSerializer(typeof(TypeWithoutPublicSetter));
+            var value = new TypeWithoutPublicSetter();
+            serializer.Serialize(ms, value);
+        }
+        catch (Exception ex)
+        {
+            Assert.Equal(ex.GetType(), typeof(InvalidOperationException));
+        }
+
+        try
+        {
+            var serializer = new XmlSerializer(typeof(TypeWithCompilerGeneratedAttributeButWithoutPublicSetter));
+            var value = new TypeWithCompilerGeneratedAttributeButWithoutPublicSetter();
+            serializer.Serialize(ms, value);
+        }
+        catch (Exception ex)
+        {
+            Assert.Equal(ex.GetType(), typeof(InvalidOperationException));
+        }
+
+        try
+        {
+            var serializer = new XmlSerializer(typeof(InvalidDerivedClass));
+            var value = new InvalidDerivedClass();
+            serializer.Serialize(ms, value);
+        }
+        catch (Exception ex)
+        {
+            Assert.Equal(ex.GetType(), typeof(InvalidOperationException));
+        }
+
+        try
+        {
+            var serializer = new XmlSerializer(typeof(AnotherInvalidDerivedClass));
+            var value = new AnotherInvalidDerivedClass();
+            serializer.Serialize(ms, value);
+        }
+        catch (Exception ex)
+        {
+            Assert.Equal(ex.GetType(), typeof(InvalidOperationException));
+        }
+
+        try
+        {
+            var serializer = new XmlSerializer(typeof(InternalTypeWithNestedPublicType.LevelData));
+            var value = new InternalTypeWithNestedPublicType.LevelData();
+            serializer.Serialize(ms, value);
+        }
+        catch (Exception ex)
+        {
+            Assert.Equal(ex.GetType(), typeof(InvalidOperationException));
+        }
+
+        try
+        {
+            var serializer = new XmlSerializer(typeof(InternalTypeWithNestedPublicTypeWithNestedPublicType.NestedPublicType.LevelData));
+            var value = new InternalTypeWithNestedPublicTypeWithNestedPublicType.NestedPublicType.LevelData();
+            serializer.Serialize(ms, value);
+        }
+        catch (Exception ex)
+        {
+            Assert.Equal(ex.GetType(), typeof(InvalidOperationException));
+        }
+
+        var value1 = new DuplicateTypeNamesTest.ns1.ClassA();
```

Comment:
Do you think we still need to keep these two variations ? Because this case is testing  **CompilationIssue**, but these variations will not throw exception, and they are normal serialization/deserialization test scenarios.